### DISCUSSION TITLE
Separate Countdown card

### DISCUSSION THREAD

**Discussion by wwwescape** [marcokreeft87/formulaone-card#262] [General]
Not related to the FormulaOne Card but... your FormulaOne Card has one of the best implementations of a countdown for Home Assistant I have seen to date.

I was wondering if you would be able to make a separate simple countdown card for use in Home Assistant based on this implementation. I'm particularly looking to use a the reverse countdown to show me how long a device has been running in real time. I'm sure that there would be others who would want such a card as well.

Hope you would consider this, if you have time to spare.

**Comment by marcokreeft87** [comment]
If you can program you could just use the code yourself. It's open source

**Reply by wwwescape** [reply]
I have been at it for days but haven't got it to work.

I'll keep trying.

Thanks.

**Reply by marcokreeft87** [reply]
Well let me know. I can probably make it for you if you don't get it to work

**Reply by wwwescape** [reply]
I finally got it to work! Thanks.

https://github.com/wwwescape/timer-card

Have credited you in the README.md but I don't plan to release it just yet.

I will probably refactor the code a bit and add more tests as I go along. I do have one issue though. When using an entity, the countdown doesn't reset when the entity's state value gets updated until I refresh the page. I haven't been able to figure this out and if you do get a chance, please let me know how I can fix it.

**Reply by marcokreeft87** [reply]
Check the shouldupdate method on this: https://github.com/marcokreeft87/room-card/blob/master/src/index.ts

**Reply by wwwescape** [reply]
Brilliant! That worked. Thanks again.

I will now try adding more tests to get close to 100% coverage. I also need to fix the GitHub Action that I copied. Pretty new area for me.

I know this code is open sourced but since I used your code, I still wanted to ask if it is okay to share this card with the Home Assistant community.

**Reply by wwwescape** [reply]
I will now use the knowledge I gained from working with your code to refactor my [sky-tonight-card](https://github.com/wwwescape/sky-tonight-card).

**Reply by marcokreeft87** [reply]
Nice! Try to get it in HACS. That will help people find it
